lib/misc.zsh

```diff
@@ -1,14 +1,19 @@
 ## Load smart urls if available
-for d in $fpath; do
-	if [[ -e "$d/url-quote-magic" ]]; then
-		if [[ -e "$d/bracketed-paste-magic" ]]; then
-			autoload -Uz bracketed-paste-magic
-			zle -N bracketed-paste bracketed-paste-magic
-		fi
-		autoload -U url-quote-magic
-		zle -N self-insert url-quote-magic
-	fi
-done
+# bracketed-paste-magic is known buggy in zsh 5.1.1 (only), so skip it there; see #4434
+autoload -Uz is-at-least
+if [[ $ZSH_VERSION != 5.1.1 ]]; then
+  for d in $fpath; do
+  	if [[ -e "$d/url-quote-magic" ]]; then
+  		if is-at-least 5.1; then
```

Comment:
More precise detection of when `bracketed-paste-magic` is needed. See [`url-quote-magic` comments](https://github.com/zsh-users/zsh/blob/a9df6aaa702abf761b155cd842a7f6917be44139/Functions/Zle/url-quote-magic#L11-L14).